### DISCUSSION TITLE
Is it possible to include a link from the YAML header metadata for a background-iframe?

### DISCUSSION THREAD

**Discussion by weberam2** [quarto-dev/quarto-cli#9891] [Q&A] [answered]
### Description

In my yaml header, let's say I create a variable: htmllink:

```
---
htmllink: "https://somelink.com"
---
```

I would like to use this link for my background-iframe options

But if I try:

```
## Slide Title {background-iframe="{{< meta htmllink >}}"
```

It doesn't work :(

Any ideas?

**Comment by cscheid** [comment]
This is a Quarto bug - it's not easy to fix, but it's 100% a bug.

**Reply by cscheid** [reply, marked as answer]
Tracking on #9903.

**Reply by cscheid** [reply]
I "fixed" this on `main`, but it will require a slightly different syntax.

In 1.5.43, the following should work:

```
## Slide Title {background-iframe='{{< meta htmllink >}}'}
```

Do note that it will require _single quotes_ instead of double quotes. That's admittedly ugly, but it's a necessary workaround for now.

**Comment by weberam2** [comment]
Hi

I pulled the latest quarto

I tried your solution, and it is different now, but the link does not open?

```
---
htmllink: https://www.quarto.org
----
```

(note, I tried without quotations, and I tried single or double quotes)

then `## Test {background-iframe='{{< meta htmllink >}}'}`

But the page looks like this now:

![image](https://github.com/quarto-dev/quarto-cli/assets/16805029/fa71f0a2-5901-42d5-8f71-ccd15ddfa75c)

**Reply by cscheid** [reply]
Ah, I'm so sorry, and thanks for following up. That's a _separate_ bug. I added a regression test but it didn't catch this instance.

What's happening is that we have a detection phase for shortcodes (for efficiency), and that phase is now not catching shortcodes inside attributes.

But if you have _any other_ shortcodes, then the detection happens, and things work.

For example, this will work:

````
---
format: revealjs
htmllink: https://www.quarto.org
---

## Test {background-iframe='{{< meta htmllink >}}'}

{{< placeholder >}}
````

Fix incoming.

**Reply by weberam2** [reply]
Oh that does work! Even using a comment makes it work:

```
## Test {background-iframe='{{< meta htmllink >}}'}

<!-- {{< placeholder >}} -->
```

Thank you!

**Reply by mcanouil** [reply]
FYI, the comment does not block shortcode evaluation (or any evaluation actually).

**Reply by cscheid** [reply]
Fix incoming https://github.com/quarto-dev/quarto-cli/pull/9924
